test/csuite/time_shift_test.sh

```diff
@@ -0,0 +1,78 @@
+#! /bin/sh
+
+set -e
+
+# uncomment the below line if this has to be run separately
+#export TESTUTIL_ENABLE_LONG_TESTS=1
+
+# We will run only when long tests are enabled.
+test "$TESTUTIL_ENABLE_LONG_TESTS" = "1" || exit 0
+
+export DONT_FAKE_MONOTONIC=1
+RUN_OS=$(uname -s)
+
+# linux we run with cpu affinity, to control the execution time
+# if we don't control the execution time this test is not effective
+echo "test read write lock for time shifting using libfaketime"
+SEC1=`date +%s`
+if [ "$RUN_OS" = "Darwin" ]
+then
+    ./test_rwlock
+elif [ "$RUN_OS" = "Linux" ]
+then
+    taskset -c 0-1 ./test_rwlock
+else
+    echo "not able to decide running OS, so exiting"
+    exit 1
+fi
+
+SEC2=`date +%s`
+DIFF1=$((SEC2 - SEC1))
+
+# preload libfaketime
+if [ "$RUN_OS" = "Darwin" ]
+then
+    export DYLD_FORCE_FLAT_NAMESPACE=y
+    export DYLD_INSERT_LIBRARIES=./libfaketime.1.dylib
```

Comment:
yes, existence of the library can be checked. Will check this only we finalize the method of adding libfaketime to jenkins.